controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobController.java

```diff
@@ -424,11 +425,18 @@ public ApplicationVersion submit(TenantAndApplicationId id, Optional<SourceRevis
                                                 sourceUrl,
                                                 revision.map(SourceRevision::commit),
                                                 false));
+            String diff = application.get().latestVersion()
+                    .map(ApplicationVersion::buildNumber)
+                    .flatMap(build -> build.isPresent() ? Optional.of(build.getAsLong()) : Optional.empty())
```

Comment:
This should always be present.

Reply:
Why not justt use getApplicationPackage from ApplicationController, with latestVersion and app id as arguments? Ah, that's the one you removedd in the first commit ... :>